drape_frontend/route_renderer.cpp

```diff
@@ -383,21 +385,44 @@ CirclesPackHandle * RouteRenderer::GetPreviewHandle(size_t & index)
   return nullptr;
 }
 
-dp::Color RouteRenderer::GetMaskColor(RouteType routeType, double baseDistance,
-                                      bool arrows) const
+// Arrow mask is used to adjust route color for subroutes after the first stop. So subroute to the first stop
+// point has default color, while subsequent subroutes color is adjusted with RouteArrowsMaskCar,
+// RouteArrowsMaskBicycle and kRouteMaskPedestrian properties.
+dp::Color RouteRenderer::GetRouteMaskColor(RouteType routeType, double baseDistance) const
 {
   if (baseDistance != 0.0 && m_distanceFromBegin < baseDistance)
   {
     if (routeType == RouteType::Car)
-      return GetColorConstant(arrows ? kRouteArrowsMaskCar : kRouteMaskCar);
+      return GetColorConstant(kRouteMaskCar);
     if (routeType == RouteType::Bicycle)
-      return GetColorConstant(arrows ? kRouteArrowsMaskBicycle : kRouteMaskBicycle);
+      return GetColorConstant(kRouteMaskBicycle);
     if (routeType == RouteType::Pedestrian)
       return GetColorConstant(kRouteMaskPedestrian);
   }
   return {0, 0, 0, 0};
 }
 
+// Arrow mask is used to adjust route arrow color. First subroute has primary arrows color
+// defined by RoutePrimaryArrowsMaskCar and RoutePrimaryArrowsMaskBicycle properties.
+dp::Color RouteRenderer::GetArrowMaskColor(RouteType routeType, double baseDistance) const
+{
+  if (baseDistance == 0.0)
+  {
+      if (routeType == RouteType::Car)
+        return GetColorConstant(kRoutePrimaryArrowsMaskCar);
+      if (routeType == RouteType::Bicycle)
+        return GetColorConstant(kRoutePrimaryArrowsMaskBicycle);
```

Comment:
kRouteFirstSegmentArrowsMaskBicycle?

Primary is not clear.

Reply:
I couldn't come up with better name. Any ideas?

Reply:
> kRouteFirstSegmentArrowsMaskBicycle?